wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryManager.java

```diff
@@ -133,8 +141,12 @@ public Factory updateFactory(Factory update, Set<FactoryImage> images) throws Co
                                                                                   ServerException {
         requireNonNull(update);
         final AuthorImpl creator = factoryDao.getById(update.getId()).getCreator();
+        FactoryImpl updateImpl = new FactoryImpl(update, images);
+        if (isNullOrEmpty(updateImpl.getName())) {
```

Comment:
In this case you cant re-assign the name to another factory. I don't know if it's good.